scene/resources/gradient.cpp

```diff
@@ -145,13 +145,21 @@ void Gradient::set_offset(int pos, const float offset) {
 	if (points.size() <= pos) {
 		points.resize(pos + 1);
 	}
+	if (!is_sorted) {
+		points.sort();
+		is_sorted = true;
+	}
```

Comment:
Could you put this block in a private method like `update_sorting` to avoid repeating it? In case some changes are needed later it will be just in one place.

Reply:
@pouleyKetchoupp I've fixed this and updated the `get_color_at_offset` function to call the new method as well.